File: scripts/volatile_and_force_compiler_task/huffman-coding/python/huffman-coding.py
from heapq import heappush, heappop, heapify
from collections import defaultdict

def encode(symb2freq):
    """this is an example for huffman encoding"""
    heap = [[wt, [sym, ""]] for sym, wt in symb2freq.items()]
    heapify(heap)
    while len(heap) > 1:
        lo = heappop(heap)
        hi = heappop(heap)
        for pair in lo[1:]:
            pair[1] = '0' + pair[1]
        for pair in hi[1:]:
            pair[1] = '1' + pair[1]
        heappush(heap, [lo[0] + hi[0]] + lo[1:] + hi[1:])
    return sorted(heappop(heap)[1:], key=lambda p: (len(p[-1]), p))

r = 1

def executeTask(i): 
    txt = "this is an example for huffman encoding" +`i`
    symb2freq = defaultdict(int)
    for ch in txt:
        symb2freq[ch] += 1
    huff = encode(symb2freq)
    return i + 1


for i in range(1000000):
 	r = executeTask(i)

File: scripts/volatile_and_force_compiler_task/huffman-coding/php/huffman-coding.php
<?php
function encode($symb2freq) {
    $heap = new SplPriorityQueue;
    $heap->setExtractFlags(SplPriorityQueue::EXTR_BOTH);
    foreach ($symb2freq as $sym => $wt)
        $heap->insert(array($sym => ''), -$wt);

    while ($heap->count() > 1) {
        $lo = $heap->extract();
        $hi = $heap->extract();
        foreach ($lo['data'] as &$x)
            $x = '0'.$x;
        foreach ($hi['data'] as &$x)
            $x = '1'.$x;
        $heap->insert($lo['data'] + $hi['data'],
                      $lo['priority'] + $hi['priority']);
    }
    $result = $heap->extract();
    return $result['data'];
}

$r = 1;

function executeTask($i) {

	$txt = 'this is an example for huffman encoding' . $i;
	$symb2freq = array_count_values(str_split($txt));
	$huff = encode($symb2freq);
	return $i + 1;
}

for ( $i = 0; $i < 1000000; $i++) {
	$r = executeTask($i + $r);
}
?>
